encoding/x/protobuf/inbound.go

```diff
@@ -78,9 +81,16 @@ func (u *unaryHandler) Handle(ctx context.Context, transportRequest *transport.R
 		}
 		responseData = protoBuffer.Bytes()
 	}
+	if isRawResponse(transportRequest.Headers) {
```

Comment:
can we add a comment explaining how this is for the decision whether we want to envelope the response or not (if that's what it is for, I'll admit I'm not 100% sure)

Would also love a task/issue that explains the issue in detail so we can have some back and forth convo on what's going on here.